### ISSUE TITLE
Unresolved symbol for macro-generated defs

### ISSUE THREAD

**Issue by l3nz** [borkdude/clj-kondo#306]
**Is your feature request related to a problem? Please describe.**

If I have a macro that generates a var at compile time, e.g. by using https://github.com/l3nz/say-cheez , Kondo complains:

`src/l2i/cfg.clj:5:23: info: unresolved symbol BUILD`

The ns creating the issue is:

```
(ns l2i.cfg
  (:require [say-cheez.core :refer [capture-build-env-to]]))

(capture-build-env-to BUILD)
```

**Describe the solution you'd like**

I guess I need a way to "mute"  Kondo on this warning? but I would not  know how to do it. Or tell it that this resolves to a def.

**Comment by borkdude** [comment]
For this case, I recommend using `(declare BUILD)`. As declaring a var that's already defined, doesn't do anything, this is a safe solution.

More information: https://github.com/borkdude/clj-kondo/blob/master/doc/config.md#exclude-unresolved-symbols-from-being-reported

**Comment by l3nz** [comment]
Come to think of it, I could add a new function that just macroexpands to the contents of the def, so there  is an explicit  def that will be slightly less elegant but will make life easier for linters :-)
